### PR TITLE
feat: re-implement resource name helper classes

### PR DESCRIPTION
**Pull request by hzyi-google**
- Delete `ParentName`, `ParentNames` and `UntypedParentName`
- Generate two overloads for methods that take `ParentName` as a parameter, changing to `LocationName` and `ProjectName` respectively
- Deprecate `LocationRunName`, `ProjectRunName`, `LocationTransferConfigName`, `ProjectTransferConfigName`, `LocationDataSourceName`, `ProjectDataSourceName`, the functionality is replaced by the new multi-pattern resource name helper classes: `RunName`, `TransferConfigName` and `DataSourceName`.
- New methods in `RunName`, `TransferConfigName` and `DataSourceName` are marked as `@betaApi`.

### CODE REVIEW

**Comment by hzyi-google** [comment]
@chingor13 Hey Jeff, how do I disable binary compatibility check?

File: google-cloud-bigquerydatatransfer/src/main/java/com/google/cloud/bigquery/datatransfer/v1/stub/DataTransferServiceStubSettings.java [hunk 1]
```diff
@@ -712,7 +712,7 @@ private static Builder initDefaults(Builder builder) {
 
       builder
           .deleteTransferConfigSettings()
-          .setRetryableCodes(RETRYABLE_CODE_DEFINITIONS.get("idempotent"))
+          .setRetryableCodes(RETRYABLE_CODE_DEFINITIONS.get("non_idempotent"))
```
Comment (hzyi-google) [inline google-cloud-bigquerydatatransfer/src/main/java/com/google/cloud/bigquery/datatransfer/v1/stub/DataTransferServiceStubSettings.java]: I checked with the API team, this change is intended behavior. `DeleteTransfer`, `StartManualTransfer` and `DeleteTransferRun` are all non-idempotent methods.

**Comment by hzyi-google** [comment]
@stephaniewang526 PTAL

File: google-cloud-bigquerydatatransfer/src/main/java/com/google/cloud/bigquery/datatransfer/v1/stub/DataTransferServiceStubSettings.java (hunk 1 above)
Comment (stephaniewang526) [reply google-cloud-bigquerydatatransfer/src/main/java/com/google/cloud/bigquery/datatransfer/v1/stub/DataTransferServiceStubSettings.java]: okay - got it

**Pull request merged**